code/game/objects/structures/janicart.dm

```diff
@@ -211,31 +229,53 @@
 /obj/structure/stool/bed/chair/janicart/relaymove(mob/user, direction)
 	if(user.stat || user.stunned || user.weakened || user.paralysis)
 		unbuckle_mob()
-	if(istype(user.l_hand, /obj/item/key) || istype(user.r_hand, /obj/item/key))
-		if(!Process_Spacemove(direction))
+	if(istype(user.l_hand, keytype) || istype(user.r_hand, keytype))
+		if(!Process_Spacemove(direction) || !has_gravity(src.loc) || move_delay)
 			return
 		step(src, direction)
 		update_mob()
 		handle_rotation()
+		if(istype(src.loc, /turf/simulated))
+			var/turf/simulated/T = src.loc
+			if(T.wet == 2)	//Lube! Fall off!
+				playsound(src, 'sound/misc/slip.ogg', 50, 1, -3)
+				buckled_mob.Stun(7)
+				buckled_mob.Weaken(7)
+				unbuckle_mob()
+				step(src, dir)
+		move_delay = 1
+		spawn(2)
+			move_delay = 0
 	else
 		user << "<span class='notice'>You'll need the keys in one of your hands to drive this [callme].</span>"
 
 
-/obj/structure/stool/bed/chair/janicart/post_buckle_mob(mob/living/M)
-	update_mob()
-	return ..()
+/obj/structure/stool/bed/chair/janicart/user_buckle_mob(mob/M, mob/user)
+	if(M != user || !ismob(M) || get_dist(src, user) > 1 || user.restrained() || user.lying || user.stat || M.buckled || istype(user, /mob/living/silicon))
+		return
 
+	unbuckle_mob()
 
-/obj/structure/stool/bed/chair/janicart/unbuckle_mob()
-	var/mob/living/M = ..()
-	if(M)
-		M.pixel_x = 0
-		M.pixel_y = 0
-	return M
+	M.visible_message(\
+		"<span class='notice'>[M] climbs onto the [callme]!</span>",\
+		"<span class='notice'>You climb onto the [callme]!</span>")
 
+	M.buckled = src

```

Comment:
yeah because I totally shouldn't expect the person who's working on the janicart to actually be a good coder capable of fixing an issue without copy paste.